core/file_api.php

```diff
@@ -98,45 +98,49 @@ function file_get_display_name( $p_filename ) {
 }
 
 /**
- * Check the number of attachments a bug has (if any)
- * @param integer $p_bug_id A bug identifier.
- * @return integer
+ * Fills the cache with the attachement count from a list of bugs
+ * using one sql query
+ * @param array $p_bug_ids Array of bug ids
  */
-function file_bug_attachment_count( $p_bug_id ) {
+function file_bug_attachment_count_cache( array $p_bug_ids ){
 	global $g_cache_file_count;
 
-	# First check if we have a cache hit
-	if( isset( $g_cache_file_count[$p_bug_id] ) ) {
-		return $g_cache_file_count[$p_bug_id];
+	if( null === $p_bug_ids || !is_array( $p_bug_ids ) ){
+		return;
 	}
 
-	# If there is no cache hit, check if there is anything in
-	#   the cache. If the cache isn't empty and we didn't have
-	#   a hit, then there are not attachments for this bug.
-	if( count( $g_cache_file_count ) > 0 ) {
-		return 0;
-	}
+	$t_sql_list= '(' . implode( ', ', $p_bug_ids ) . ')';
+	$t_query = 'SELECT B.id AS bug_id, COUNT(F.bug_id) AS attachments'
+			. ' FROM {bug} B LEFT OUTER JOIN {bug_file} F'
+			. ' ON (B.id = F.bug_id)'
+			. ' WHERE B.id IN ' . $t_sql_list . ' GROUP BY B.id';
 
```

Comment:
> There is a limit in how many items can fit in (especially with Oracle DB), which will lead to errors when executing the code on large data sets

Right.
This is already an issue with bugnote count (see #696), with the current functions.
And probably more places: retrieving projects, cache bug rows, etc...

In standard usage (with current code base): bug listings, are limited to pagination size, so this is not practical problem (but still it exists and waiting to bite us is some corner cases)
A real use case where a massive enough bug list may trigger the fail situation would be CSV/Excel exports, which currently has disabled the inclusion of these two fields (again, #696)

So even if this would not result in fail, with current use cases, I agree to fix first the cache functions to address the long IN failure
So in order to get this done, i'd like some feedback on what proposed methods would be convenient to use, to provide a uniform approach on other similar situations. Brainstorming, some possibilities:
- loop through the array partitioned in pieces of 500 item size, performing ~N/500 queries. (or any other convenient chunk size)
- use a temporary table to store values and join.
- ¿any other?